rest/oidc_api.go

```diff
@@ -31,6 +31,31 @@ const (
 	requestParamState        = "state"
 	requestParamScope        = "scope"
 	requestParamRedirectURI  = "redirect_uri"
+
+	// stateCookieName is the name of state cookie to prevent cross-site request forgery (CSRF).
+	stateCookieName = "sg-oidc-state"
+
+	// stateCookieTimeout represents the duration of state cookie; state
+	// cookie expires within 5 minutes.
+	stateCookieTimeout = time.Minute * 5
+)
+
+var (
+	// ErrNoStateCookie is returned by handler's handleOIDCCallback method when
+	// the state cookie is not found during OpenID Connect Auth callback.
+	ErrNoStateCookie = base.HTTPErrorf(http.StatusBadRequest,
+		"OIDC Auth Failure: No state cookie found; possible request forgery")
```

Comment:
'possible request forgery' here and below feels like it's guessing at a root cause that we don't necessarily have evidence for.  I'd prefer to just have these errors state the nature of the error from Sync Gateway's perspective (no cookie found, state mismatch, etc)